File: 2390-removing-stars-from-a-string/2390-removing-stars-from-a-string.js
/**
 * @param {string} s
 * @return {string}
 */
var removeStars = function(s) {
    
    const strs = [];
    for(let i = 0; i<s.length; i++){
        
        if(s.charAt(i) === '*'){
            strs.pop();
        }else{
            strs.push(s.charAt(i))
        }
    }
    return strs.join('');
};

File: 2390-removing-stars-from-a-string/2390-removing-stars-from-a-string.cpp
class Solution {
public:
    string removeStars(string s) {
        
        stack<char>st;
        for(int i = 0; i<s.size(); i++){
            if(s[i] == '*'){
                st.pop();
            }else{
                st.push(s[i]);
            }
        }
        
        string str = "";
        while(!st.empty()){
            str += st.top();
            st.pop();
        }
        
        reverse(str.begin(),str.end());
        return str;
    }
};